api/src/main/java/com/datatorrent/api/Module.java

```diff
@@ -21,8 +21,132 @@
 import org.apache.hadoop.classification.InterfaceStability;
 import org.apache.hadoop.conf.Configuration;
 
+import com.datatorrent.api.Context.PortContext;
+import com.datatorrent.api.Operator.InputPort;
+import com.datatorrent.api.Operator.OutputPort;
+import com.datatorrent.api.Operator.Port;
+import com.datatorrent.api.Operator.Unifier;
+
 @InterfaceStability.Evolving
 public interface Module
 {
   void populateDAG(DAG dag, Configuration conf);
+
+  public interface ProxyPort<T> extends Port
```

Comment:
Yes, this was not being used anywhere, ProxyInputPort and ProxyOutputPort were extending it, but this is not actually needed. I have remove it.